lib/internal/abort_controller.js

```diff
@@ -258,6 +271,7 @@ class AbortSignal extends EventTarget {
         resultSignal[kSourceSignals].add(signalWeakRef);
         signal[kDependantSignals].add(resultSignalWeakRef);
         dependantSignalsCleanupRegistry.register(resultSignal, signalWeakRef);
+        finalizer.register(signal, { sourceSignalRef: signalWeakRef, composedSignalRef: resultSignalWeakRef });
```

Comment:
This should also happen down below in the else branch, right? Basically everywhere where `dependantSignalsCleanupRegistry` is used, `finalizer` should be used as well, since it's kind of the inverse.
Or am I missing something?

Reply:
The `else` is for composed signals. So when it comes to there, it will already have the finalizer in its source signals.

That's a good test case to have. Let me try adding it so it will be clear.

Reply:
AFAICT, the else case with the composed signal is pretty much the same as if all the source signals in the composed signal were passed directly to us.
Of course the composed signal has its own finalizers registered. But these ones only clean up the composed signal itself, not us.
For instance, in the case that we are created and passed only one composed signal, and all source signals of the composed signal are finalized, the composed signal itself is also finalized, but we don't notice that and so we can't notice that we can't be aborted and so the signal handlers cannot be cleaned up.
Or am I missing something here?

Reply:
It makes a lot of sense. Thanks for pointing that out.

Reply:
100% correct. When I added the test it kept the composed one forever. I just pushed the fix and updated the test to handle this case.